scripts/components/CalendarAddEventForm.jsx

```diff
@@ -0,0 +1,167 @@
+
+    this.setState({
+      validationError: undefined,
+    });
+
+    var validation = this.validate();
+    if (validation.error) {
+      this.setState({ validationError: validation.error });
+      return;
+    }
+
+    Api.addPropertyCalendarEvents({
+      data: {
+        eventDesc: this.state.eventDesc,
+        date: this.state.date,
+        time: this.state.time,
+        notes: this.state.notes,
+        propertyId: propertyId,
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
+        }
+
+        // Clear the form
+        this.setState({
+          eventDesc: '',
+          date: '',
+          time: '',
+          notes: '',
+        });
+
+        this.props.eventAdded();
+        this.refs.dialog.dismiss();
+      }
+    });
+  },
+
+  // Validate the form, returns the Joi result of the validation.
+  validate() {
+    return Joi.validate({
+      'Event Description': this.state.eventDesc,
+      Date: this.state.date,
+      Time: this.state.time,
+      Notes: this.state.notes,
+    }, schema);
+   },
+
+  render() {
+    var { eventDesc, date, time, notes, validationError } = this.state;
+    var standardActions = [
+      { text: 'Cancel' },
+      { text: 'Add Event', onTouchTap: this.onSubmit, ref: 'submit' }
+    ];
+    var validationErrorMsg = validationError ? (
+      <JoiError error={validationError} fillBackground={true} />
+    ) : null;
+
+    return (
+      <div style={style.formContainer}>
+        <RaisedButton label="Add Event"
+                      primary={true}
+                      onClick={this.onButtonClick} />
+        <Dialog
+          title="Add an Event"
+          actions={standardActions}
+          actionFocus="submit"
+          modal="true"
+          ref="dialog">
+
+          {validationErrorMsg}
+
+          <TextField
+            value={eventDesc}
+            name="Event Description"
+            onChange={this.onChange.bind(this, 'eventDesc')}
+            floatingLabelText="Event Description"
+            hintText="Describe what will be happening" />
+          <br />
+          <TextField
+            value={date}
+            name="Date"
+            onChange={this.onChange.bind(this, 'date')}
+            floatingLabelText="Date"
+            hintText="Enter the date. (dd/mm/yyyy)" />
+          <TextField
+            value={time}
+            name="Time"
+            onChange={this.onChange.bind(this, 'time')}
+            floatingLabelText="Time"
+            hintText="Enter the time. (h:mm ampm)" />
+          <TextField
+            value={notes}
+            multiLine={true}
+            name="Notes"
+            onChange={this.onChange.bind(this, 'notes')}
+            floatingLabelText="Notes"
+            hintText="Add more detailed notes here"
+            fullWidth />
+        </Dialog>
+      </div>
+    );
+  }
+});
+
+var style = {
+
+};
+
+// Validation schema for user profile form data.
```

Comment:
Be careful when copy pasting code from other places that you don't leave obsolete comments, as this can lead to lots of confusion.